### DISCUSSION TITLE
Dynamic Target Rate configuration update

### DISCUSSION THREAD

**Discussion by davria18** [BottlecapDave/HomeAssistant-OctopusEnergy#599] [Ideas]
Hi

Is it, or would it be possible to dynamically update the Target Rate configuration variables based on other data held in Home Assistant?

My use case is that I would like to increase the number of hours included in the target rates as the outside temperature gets colder, so that my heating comes on more often but still uses the cheapest slots available.  I was thinking that I could run an automation at midnight to increase the number of hours in each target rate time period based on the forecast low for the next day.

I hope that is clear (it seems it in my head).

Many thanks

David

**Comment by BottlecapDave** [comment]
Hello. What you're after is the [update config service](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/services.md#octopus_energyupdate_target_config). Please note, that these changes are not persistented between HA restarts or integration reloads.

**Comment by davria18** [comment]
Thank youRegardsDavidOn 4 Dec 2023, at 17:03, David Kendall ***@***.***> wrote:﻿
Hello. What you're after is the update config service. Please note, that these changes are not persistented between HA restarts or integration reloads.

**Comment by davria18** [comment]
Hi Dave

How will I know if the configuration has been updated.  I am running the automation below but when I run the automation and then go back into the Update Target Rate configuration screen the number of hours has not changed.  Is this the correct behaviour or should I expect it to change?

alias: "Heating: Set number of low rate costs dependant on outside temperature"
trigger:
  - platform: time_pattern
    minutes: "58"
  - platform: time_pattern
    minutes: "28"
condition: []
action:
  - if:
      - condition: numeric_state
        entity_id: sensor.openweathermap_temperature
        below: 0
    then:
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1.5"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_0000_to_0400
      - service: octopus_energy.update_target_config
        data:
          target_hours: "2.5"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_0400_to_0800
      - service: octopus_energy.update_target_config
        data:
          target_hours: "2"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_0800_to_1200
      - service: octopus_energy.update_target_config
        data:
          target_hours: "2"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_1200_to_1600
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1.5"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_1900_to_2200
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_2200_to_0000
  - if:
      - condition: numeric_state
        entity_id: sensor.openweathermap_temperature
        below: 5
        above: 0
    then:
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1.5"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_0000_to_0400
      - service: octopus_energy.update_target_config
        data:
          target_hours: "2"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_0400_to_0800
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1.5"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_0800_to_1200
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1.5"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_1200_to_1600
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1.5"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_1900_to_2200
      - service: octopus_energy.update_target_config
        data:
          target_hours: "1"
        target:
          entity_id: binary_sensor.octopus_energy_target_heating_2200_to_0000

Many thanks

Dave

**Reply by BottlecapDave** [reply]
The config will still have the old settings (at the time I couldn't find a way of updating these permanently). However you can verify the change by looking at the attributes of the sensor.
